src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -131,15 +133,31 @@ private void initializeHandlers() {
     }
 
     private void subscribeToConfigChanges() {
+        // This first call is just to initialize an empty configuration
         onConfigurationChanged();
         config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this::configChangeHandler);
     }
 
     private void onConfigurationChanged() {
-        try {
-            cdaConfiguration = CDAConfiguration.from(cdaConfiguration, getConfig());
-        } catch (URISyntaxException e) {
-            serviceErrored(e);
+        Result<CDAConfiguration> result = CDAConfiguration.from(cdaConfiguration, getConfig());
+        CDAConfiguration configuration = result.get();
+
+        // Nothing has changed
+        if (configuration.hasChanged(cdaConfiguration)) {
+            return;
+        }
+
+        cdaConfiguration = configuration;
+
+        if (result.isOk() && inState(State.BROKEN)) {
```

Comment:
If the new configuration we are getting is ok but we were broken due to a bad config we try to re install.